presto-docs/src/main/sphinx/connector/base-arrow-flight.rst

```diff
@@ -0,0 +1,106 @@
+
+======================
+Arrow-flight Connector
+======================
+This connector allows querying multiple data sources that are supported by an Arrow Flight server.
+Apache Arrow enhances performance and efficiency in data-intensive applications through its columnar memory layout, zero-copy reads, vectorized execution, cross-language interoperability, rich data type support, and optimization for modern hardware. These features collectively reduce overhead, improve data processing speeds, and facilitate seamless data exchange between different systems and languages.
+
+Getting Started with base-arrow-module: Essential Abstract Methods for Developers
+--------------
+To utilize the base-arrow-module, you need to implement certain abstract methods that are specific to your use case. Below are the required classes and their purposes:
+
+* ``ArrowFlightClientHandler.java``
+  This class is responsible for initializing the Flight client and retrieving Flight information from the Flight server. To authenticate the Flight server, you must implement the abstract method ``getCallOptions`` in ArrowFlightClientHandler, which returns the ``CredentialCallOption`` specific to your Flight server.
+
+* ``ArrowAbstractFlightRequest.java``
+  Implement this class to define the request data, including the data source type, connection properties, the number of partitions and other data required to interact with database.
+
+* ``ArrowAbstractMetadata.java``
+  To retrieve metadata (schema and table information), implement the abstract methods in the ArrowAbstractMetadata class.
+
+* ``ArrowAbstractSplitManager.java``
+  Extend the ArrowAbstractSplitManager class to implement the Arrow flight request, defining the Arrow split.
+
+* ``ArrowPlugin.java``
+  Register your connector name by extending the ArrowPlugin class.
+
+Configuration
+-------------
+To configure the Arrow connector, create a catalog file
+in ``etc/catalog`` named, for example, ``arrowmariadb.properties``, to
+mount the Arrow-flight connector as the ``arrowmariadb`` catalog.
+Create the file with the following contents, replacing the
+connection properties as appropriate for your setup:
+
+
+.. code-block:: none
+
+
+        connector.name=<CONNECTOR_NAME> 
+
+        arrow-flight.server=<SERVER_ENDPOINT>
+        arrow-flight.server.port=<SERVER_PORT>
+
+        data-source.name=<DATASOURCE_NAME>
+        data-source.host=<HOST>
+        data-source.database=<DATABASE_NAME>
+        data-source.username=<USERNAME>
+        data-source.password=<PASSWORD>
+        data-source.port=<PORT>
+        data-source.ssl=<TRUE/FALSE>
+
+        Add other properties that are required for your Flight server to connect.
```

Comment:
```suggestion
Add other properties that are required for your Flight server to connect.
```
Formatting suggestion, because this sentence does not appear to be part of the code block.